lib/SILGen/SILGenStmt.cpp

```diff
@@ -74,7 +74,9 @@ void SILGenFunction::eraseBasicBlock(SILBasicBlock *block) {
   assert(block->pred_empty() && "erasing block with predecessors");
   assert(block->empty() && "erasing block with content");
   if (block == StartOfPostmatter) {
-    StartOfPostmatter = &*std::next(block->getIterator());
+    StartOfPostmatter = (std::next(block->getIterator()) == F.end())
+                            ? nullptr
+                            : &*std::next(block->getIterator());
```

Comment:
This `std::next or null` pattern comes up twice in the patch. It might be a good idea to factor this into a `next_or_null(iterator, end)` helper. That would allow the change to `SpeculativeDevirtualizer` to be simplified as well, using `dyn_cast_or_null<StrongReleaseInst>(next_or_null(Continue->begin(), Continue->end())` instead of a nested conditional.